docs/source/guide/rem-4-low-level.md

```diff
@@ -12,3 +12,128 @@ kernelspec:
+observable = Observable(PauliString("ZI"), PauliString("IZ"))
+
+print(circuit)
+```
+
+Let's create a noisy readout executor to have erroneous values to work with:
+
+```{code-cell} ipython3
+from functools import partial
+
+import numpy as np
+from cirq.experiments.single_qubit_readout_calibration_test import (
+    NoisySingleQubitReadoutSampler,
+)
+
+from mitiq._typing import MeasurementResult
+from mitiq.raw import execute as raw_execute
+
+def noisy_readout_executor(
+    circuit, p0: float = 0.01, p1: float = 0.01, shots: int = 8192
+) -> MeasurementResult:
+    # Replace with code based on your frontend and backend.
+    simulator = NoisySingleQubitReadoutSampler(p0, p1)
+    result = simulator.run(circuit, repetitions=shots)
+
+    return MeasurementResult(
+        result=np.column_stack(list(result.measurements.values())),
+        qubit_indices=tuple(
+            # q[2:-1] is necessary to convert "q(number)" into "number"
+            int(q[2:-1])
+            for k in result.measurements.keys()
+            for q in k.split(",")
+        ),
+    )
+
+# Use a noisy executor that has a 25% chance of flipping
+p_flip = 0.25
+noisy_executor = partial(noisy_readout_executor, p0=p_flip, p1=p_flip)
+```
+
+With our noisy executor we can obtain the raw measurement results with {func}`Executor._run()`:
+
+```{code-cell} ipython3
+from mitiq.executor.executor import Executor
+
+executor = Executor(noisy_executor)
+result = executor._run([circuit])
+noisy_result = result[0]
+print(noisy_result)
+```
+
+## Mitigating the measurements with an inverse confusion matrix
+
+For this example we use a simple, uncorrelated readout error model where each qubit has a separate probability of 
+flipping from $|0\rangle \rightarrow |1\rangle$ and $|1\rangle \rightarrow |0\rangle$ during readout. More details
+of constructing this inverse confusion matrix are presented in 
+[What additional options are available when using REM?](rem-3-options.myst).
+
+```{code-cell} ipython3
+from mitiq.rem import generate_inverse_confusion_matrix
+
+inverse_confusion_matrix = generate_inverse_confusion_matrix(2, p_flip, p_flip)
+```
+
+Next, we utilize our noisy results and the inverse confusion matrix in order to obtain mitigated results:
+
+```{code-cell} ipython3
+from mitiq.rem.inverse_confusion_matrix import mitigate_measurements
+
+mitigated_result = mitigate_measurements(noisy_result, inverse_confusion_matrix)
+```
+
+The details of how the measurements are mitigated are presented in 
+[What is the theory behind REM?](rem-5-theory.myst).
+
+## Taking the expectation from the mitigated measurements
+
+Finally, once we have our mitigated results we can obtain the expectation value from the observable:
+
+```{code-cell} ipython3
+print(observable._expectation_from_measurements([mitigated_result]))
+```
+
+Based on the observable we defined, $Z_0 + Z_1$, and the circuit we are close to the expectation value $-2$.
```

Comment:
Using the private method `_expectation_from_measurements` is a bit dangerous since it assumes a particular structure for the executed circuits. E.g. change of basis for non-diagonal observables etc..

What about if we replace this section with with a histogram comparing ideal, unmitigated and mitigated measurement counts ? The method `MeasurementResult.get_counts()` can be useful.

Alternatively an expectation value such as $tr[\rho |{\rm correctbitstring} \rangle \langle {\rm correctbitstring}| ]$  ( frequency of the ideal correct bitstring) can be manually computed. Manual evaluation is fine here since we want to show a low-level way of applying REM.

Reply:
I've made the suggested changes. Thanks.